src/test/java/com/teragrep/pth10/TeragrepTransformationTest.java

```diff
@@ -581,4 +583,100 @@ public void tgGetArchiveSummaryTest() {
             Assertions.assertEquals(Arrays.asList("1", "2"), listOfResult);
         });
     }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void tgRegexExtractTokensTest() {
+        // source: "127.4.4.4"
+        String regex = "\\d+";
+        Pattern pattern = Pattern.compile(regex);
+        streamingTestUtil
+                .performDPLTest(
+                        "index=index_A | teragrep exec regexextract regex " + regex + " input source output strTokens",
+                        testFile, ds -> {
+                            List<String> result = ds
+                                    .select("strTokens")
+                                    .first()
+                                    .getList(0)
+                                    .stream()
+                                    .map(Object::toString)
+                                    .collect(Collectors.toList());
+                            Assertions.assertEquals(4, result.size());
+                            Assertions.assertTrue(result.stream().allMatch(s -> pattern.matcher(s).matches()));
```

Comment:
I'm thinking this test could go through even if the tokens that are extracted are wrong, like only 127 or only 4. Can you assert each token in the list one by one? You can order the dataset with .orderBy("offset") or something similar if the order changes (Spark does that sometimes).

Reply:
Added equals check between expected list and result list